.clang-tidy

```diff
@@ -28,4 +28,25 @@ Checks:          '-*,
                   '
 #WarningsAsErrors: '*'
 AnalyzeTemporaryDtors: false
+# Use Google style guide.
+CheckOptions:
+# Classes / Structs / Enums are CamelCase.
+  - { key: readability-identifier-naming.ClassCase,               value: CamelCase  }
+  - { key: readability-identifier-naming.StructCase,              value: CamelCase  }
+  - { key: readability-identifier-naming.EnumCase,                value: CamelCase  }
+# Variables / Parameters are lower_case.
+  - { key: readability-identifier-naming.VariableCase,            value: lower_case }
+  - { key: readability-identifier-naming.ParameterCase,           value: lower_case }
+  - { key: readability-identifier-naming.PublicMemberCase,        value: lower_case }
+# Constexpr / const things are kCamelCase.
+  - { key: readability-identifier-naming.ConstexprVariablePrefix, value: "k" }
+  - { key: readability-identifier-naming.ConstexprVariableCase,   value: CamelCase }
+  - { key: readability-identifier-naming.ClassConstantPrefix, value: "k" }
+  - { key: readability-identifier-naming.ClassConstantCase,   value: CamelCase }
+  - { key: readability-identifier-naming.GlobalConstantPrefix, value: "k" }
+  - { key: readability-identifier-naming.GlobalConstantCase,   value: CamelCase }
+# Functions are CamelCase.
+  - { key: readability-identifier-naming.FunctionCase,            value: CamelBack  }
```

Comment:
Ideally either `CamelCase` or `lower_case` rather than `CamelBack`, right? I think I personally have a _slight_ preference towards `lower_case` but we should probably come to consensus.

Reply:
lol the comment says CamelCase

Reply:
If we're following google then I think it makes sense to go google all the way and do `CamelCase`.